sink.go

```diff
@@ -24,6 +24,11 @@ type MetricSink interface {
 	AddSampleWithLabels(key []string, val float32, labels []Label)
 }
 
+type PrecisionGaugeMetricSink interface {
```

Comment:
I realise the rest of this library does a less-than-ideal job of documenting interfaces already, but I'd be inclined to add a doc comment to explain the purpose of this additional interface. Could be a one liner, could refer to the history of not originally supporting 64bit floats, could even link to this PR for more info but it's probably useful context to a future reader who wonders why this is different to MetricSink!

I really like the name, but without any comment I might assume "precision gauge" is a specific concept/semantic I wasn't aware of and wonder where to look for more info. Just a simple line or two to the effect that not all sinks may support it but if 64 bit precisions is needed this allows that would save me time looking up commit history etc!